--- security/nss/lib/pki1/oiddata.h.orig	2002-01-04 06:22:07.000000000 +0100
+++ security/nss/lib/pki1/oiddata.h	2007-07-14 16:06:23.000000000 +0200
@@ -43,7 +43,7 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
+extern const NSSOID *nss_builtin_oids;
 extern const PRUint32 nss_builtin_oid_count;
 
 /*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
