Index: extensions.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/mozapps/extensions/content/extensions.js,v
retrieving revision 1.42
diff -u -r1.42 extensions.js
--- toolkit/mozapps/extensions/content/extensions.js	8 Jan 2005 00:43:45 -0000	1.42
+++ toolkit/mozapps/extensions/content/extensions.js	29 Jan 2005 02:49:39 -0000
@@ -570,7 +570,7 @@
       
       var data = { }, length = { };
       xfer.getTransferData("text/x-moz-url", data, length);
-      var fileURL = data.value.QueryInterface(Components.interfaces.nsISupportsString).data;
+      var fileURL = data.value.QueryInterface(Components.interfaces.nsISupportsString).data.split("\n")[0];
 
       var fileURI = this._ioServ.newURI(fileURL, null, null);
       var url = fileURI.QueryInterface(Components.interfaces.nsIURL);
@@ -599,7 +599,7 @@
       
       var data = { }, length = { };
       xfer.getTransferData("text/x-moz-url", data, length);
-      var fileURL = data.value.QueryInterface(Components.interfaces.nsISupportsString).data;
+      var fileURL = data.value.QueryInterface(Components.interfaces.nsISupportsString).data.split("\n")[0];
       var uri = Components.classes["@mozilla.org/network/standard-url;1"]
                           .createInstance(Components.interfaces.nsIURI);
       uri.spec = fileURL;
