Index: os2inrval.c
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/src/md/os2/os2inrval.c,v
retrieving revision 3.6.2.2
diff -u -r3.6.2.2 os2inrval.c
--- nsprpub/pr/src/md/os2/os2inrval.c	28 Feb 2003 02:44:25 -0000	3.6.2.2
+++ nsprpub/pr/src/md/os2/os2inrval.c	12 Jan 2005 13:16:22 -0000
@@ -47,8 +47,17 @@
 void
 _PR_MD_INTERVAL_INIT()
 {
-    ULONG timerFreq = 0; /* OS/2 high-resolution timer frequency in Hz */
-    APIRET rc = DosTmrQueryFreq(&timerFreq);
+    char *envp;
+    ULONG timerFreq;
+    APIRET rc;
+
+    if ((envp = getenv("NSPR_OS2_NO_HIRES_TIMER")) != NULL) {
+        if (atoi(envp) == 1)
+           return;
+    }
+
+    timerFreq = 0; /* OS/2 high-resolution timer frequency in Hz */
+    rc = DosTmrQueryFreq(&timerFreq);
     if (NO_ERROR == rc) {
         useHighResTimer = PR_TRUE;
         PR_ASSERT(timerFreq != 0);
