Index: gfx/src/os2/nsGfxFactoryOS2.cpp
===================================================================
RCS file: /cvsroot/mozilla/gfx/src/os2/nsGfxFactoryOS2.cpp,v
retrieving revision 1.40
diff -u -r1.40 nsGfxFactoryOS2.cpp
--- gfx/src/os2/nsGfxFactoryOS2.cpp	22 Nov 2004 18:21:53 -0000	1.40
+++ gfx/src/os2/nsGfxFactoryOS2.cpp	2 Feb 2005 22:51:18 -0000
@@ -92,10 +92,6 @@
                                 KEY_ENUMERATE_SUB_KEYS | KEY_NOTIFY
 #define ERROR_SUCCESS           0L
 
-LONG _System RegOpenKeyEx(HKEY, const char*, ULONG, ULONG, HKEY* );
-LONG _System RegQueryValueEx(HKEY, const char*, ULONG*, ULONG*, UCHAR*,
-                             ULONG*);
-
 static PRBool
 UseFTFunctions()
 {
@@ -111,21 +107,45 @@
       return PR_FALSE;
     }
 
+    // Test for availability of registry functions and query their addresses
+    APIRET rc;
+    HMODULE hmod = NULLHANDLE;
+    char LoadError[CCHMAXPATH];
+    rc = DosLoadModule(LoadError, CCHMAXPATH, "REGISTRY", &hmod);
+    if (rc != NO_ERROR) {
+      NS_WARNING("REGISTRY.DLL could not be loaded");
+      return PR_FALSE;
+    }
+    LONG _System (*APIENTRY RegOpenKeyEx)(HKEY, const char*, ULONG, ULONG,
+                                           HKEY* );
+    LONG _System (*APIENTRY RegQueryValueEx)(HKEY, const char*, ULONG*, ULONG*,
+                                              UCHAR*, ULONG*);
+
+    rc = DosQueryProcAddr(hmod, 0L, "RegOpenKeyExA", (PFN*)&RegOpenKeyEx);
+    rc += DosQueryProcAddr(hmod, 0L, "RegQueryValueExA", (PFN*)&RegQueryValueEx);
+    if (rc != NO_ERROR) {
+      NS_WARNING("Registry function(s) were not found in REGISTRY.DLL");
+      DosFreeModule(hmod);
+      return PR_FALSE;
+    }
+
     // Is FT2LIB enabled?
     HKEY key;
-    LONG result = ::RegOpenKeyEx(HKEY_CURRENT_USER,
-                                 "Software\\Innotek\\InnoTek Font Engine", 0,
-                                 KEY_READ, &key);
+    LONG result = RegOpenKeyEx(HKEY_CURRENT_USER,
+                               "Software\\Innotek\\InnoTek Font Engine", 0,
+                               KEY_READ, &key);
     if (result != ERROR_SUCCESS) {
+      DosFreeModule(hmod);
       return PR_FALSE;
     }
 
     ULONG value;
     ULONG length = sizeof(value);
-    result = ::RegQueryValueEx(key, "Enabled", NULL, NULL, (UCHAR*)&value,
-                               &length);
+    result = RegQueryValueEx(key, "Enabled", NULL, NULL, (UCHAR*)&value,
+                             &length);
     if (result != ERROR_SUCCESS || value == 0) {
       // check if "Innotek Font Engine" is disabled (value == 0)
+      DosFreeModule(hmod);
       return PR_FALSE;
     }
 
@@ -139,20 +159,24 @@
     strcpy(keystr, "Software\\Innotek\\InnoTek Font Engine\\Applications\\");
     strcat(keystr, name);
     strcat(keystr, ext);
-    result = ::RegOpenKeyEx(HKEY_CURRENT_USER, keystr, 0, KEY_READ, &key);
+    result = RegOpenKeyEx(HKEY_CURRENT_USER, keystr, 0, KEY_READ, &key);
     if (result != ERROR_SUCCESS) {
+      DosFreeModule(hmod);
       return PR_FALSE;
     }
-    result = ::RegQueryValueEx(key, "Enabled", NULL, NULL, (UCHAR*)&value,
-                               &length);
+    result = RegQueryValueEx(key, "Enabled", NULL, NULL, (UCHAR*)&value,
+                             &length);
     if (result != ERROR_SUCCESS || value == 0) {
       // check if FT2LIB is disabled for our application (value == 0)
+      DosFreeModule(hmod);
       return PR_FALSE;
     }
 
+    // REGISTRY.DLL use ends here
+    DosFreeModule(hmod);
+
     // Load lib and functions
-    HMODULE hmod = 0;
-    int rc = DosLoadModule(NULL, 0, "FT2LIB", &hmod);
+    rc = DosLoadModule(LoadError, 0, "FT2LIB", &hmod);
     if (rc == NO_ERROR) {
       rc = DosQueryProcAddr(hmod, 0, "Ft2EnableFontEngine",
                             (PFN*)&nsFontMetricsOS2FT::pfnFt2EnableFontEngine);
Index: gfx/src/os2/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/gfx/src/os2/Makefile.in,v
retrieving revision 1.41
diff -u -r1.41 Makefile.in
--- gfx/src/os2/Makefile.in	9 Dec 2004 19:28:01 -0000	1.41
+++ gfx/src/os2/Makefile.in	2 Feb 2005 22:51:18 -0000
@@ -95,7 +95,6 @@
 		$(MOZ_UNICHARUTIL_LIBS) \
 		$(MOZ_JS_LIBS) \
 		$(MOZ_COMPONENT_LIBS) \
-		-lpmwinx \
 		$(NULL)
 
 GARBAGE += $(SHARED_LCPPSRCS) $(wildcard *.$(OBJ_SUFFIX))
