Index: widget/src/os2/nsDragService.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/os2/nsDragService.cpp,v
retrieving revision 1.39
diff -p -u -r1.39 nsDragService.cpp
--- widget/src/os2/nsDragService.cpp	28 Mar 2005 16:33:50 -0000	1.39
+++ widget/src/os2/nsDragService.cpp	17 Sep 2005 16:32:56 -0000
@@ -1319,15 +1319,43 @@ NS_IMETHODIMP nsDragService::NativeDataT
       rv = NS_OK;
     }
 
-      // construct the HTML flavor
+      // construct the HTML flavor - for supported graphics,
+      // use an IMG tag, for all others create a link
     nsCOMPtr<nsISupportsString> htmlPrimitive(
                             do_CreateInstance(NS_SUPPORTS_STRING_CONTRACTID));
     if (htmlPrimitive ) {
-      nsString htmlStr(NS_LITERAL_STRING("<a href=\"") +
+      nsString htmlStr;
+      nsCOMPtr<nsIURI> uri;
+
+      rv = NS_ERROR_FAILURE;
+      if (NS_SUCCEEDED(NS_NewURI(getter_AddRefs(uri), pszText))) {
+        nsCOMPtr<nsIURL> url (do_QueryInterface(uri));
+        if (url) {
+          nsCAutoString extension;
+          url->GetFileExtension(extension);
+          if (!extension.IsEmpty()) {
+            if (!stricmp(extension.get(), "gif") ||
+                !stricmp(extension.get(), "jpg") ||
+                !stricmp(extension.get(), "png") ||
+                !stricmp(extension.get(), "jpeg"))
+              rv = NS_OK;
+          }
+        }
+      }
+
+      if (NS_SUCCEEDED(rv))
+        htmlStr.Assign(NS_LITERAL_STRING("<img src=\"") +
+                       outText +
+                       NS_LITERAL_STRING("\" alt=\"") +
+                       outTitle +
+                       NS_LITERAL_STRING("\"/>") );
+      else
+        htmlStr.Assign(NS_LITERAL_STRING("<a href=\"") +
                        outText +
                        NS_LITERAL_STRING("\">") +
                        (outTitle.IsEmpty() ? outText : outTitle) +
                        NS_LITERAL_STRING("</a>") );
+
       htmlPrimitive->SetData(htmlStr);
       trans->SetTransferData(kHTMLMime, htmlPrimitive, 2*htmlStr.Length());
       rv = NS_OK;
