--- mozilla/composer/base/jar.mn.orig	2007-08-29 02:54:25.000000000 +0200
+++ mozilla/composer/base/jar.mn	2007-08-29 03:24:58.000000000 +0200
@@ -8,6 +8,7 @@
 +	content/editor/EdColorPicker.xul                         (content/colourPicker.xul)
 +	content/editor/EdColorPicker.js                          (content/colourPicker.js)
 +	content/editor/about-logo.png                            (content/about-logo.png)
++	content/editor/about-footer.png                            (content/about-footer.png)
 *+	content/editor/mainCore.js                               (content/mainCore.js)
 +	content/editor/editor.js                                 (content/editor.js)
 +	content/editor/editorRulers.js                           (content/editorRulers.js)
@@ -56,11 +57,6 @@
 +	content/editor/Validator.js                              (dialogs/Validator.js)
 +	content/editor/EdTableProps.xul                          (dialogs/EdTableProps.xul)
 +	content/editor/EdTableProps.js                           (dialogs/EdTableProps.js)
-+     content/editor/about-credits.png                         (content/about-credits.png)
-+     content/editor/about-footer.png                          (content/about-footer.png)
-+     content/editor/about-header.png                          (content/about-header.png)
-+     content/editor/linspire.gif                              (content/linspire.gif)
-+     content/editor/DI.png                                    (content/DI.png)
 +	content/editor/images/v0.png                             (images/v0.png)
 +	content/editor/images/v1.png                             (images/v1.png)
 +	content/editor/images/v2.png                             (images/v2.png)
--- mozilla/themes/classic/jar.mn.orig	2007-08-29 02:43:30.000000000 +0200
+++ mozilla/themes/classic/jar.mn	2007-08-29 02:44:12.000000000 +0200
@@ -49,10 +49,6 @@
   skin/classic/communicator/sidebar/sidebar.css                         (communicator/sidebar/sidebar.css)
   skin/classic/communicator/sidebar/sidebarListView.css                 (communicator/sidebar/sidebarListView.css)
   skin/classic/communicator/xpinstall/xpinstall.css                     (communicator/xpinstall/xpinstall.css)
-  skin/classic/communicator/brand/throbber-single.gif                   (communicator/brand/throbber-single.gif)
-  skin/classic/communicator/brand/throbber-anim.gif                     (communicator/brand/throbber-anim.gif)
-  skin/classic/communicator/brand/throbber16-single.gif                 (communicator/brand/throbber16-single.gif)
-  skin/classic/communicator/brand/throbber16-anim.gif                   (communicator/brand/throbber16-anim.gif)
   skin/classic/communicator/icons/offline.gif                           (communicator/icons/offline.gif)
   skin/classic/communicator/icons/online.gif                            (communicator/icons/online.gif)
   skin/classic/communicator/icons/search-act.gif                        (communicator/icons/search-act.gif)
@@ -62,7 +58,6 @@
   skin/classic/communicator/icons/lock-broken.gif                       (communicator/icons/lock-broken.gif)
   skin/classic/communicator/icons/lock-insecure.gif                     (communicator/icons/lock-insecure.gif)
   skin/classic/communicator/icons/loading.gif                           (communicator/icons/loading.gif)
-  skin/classic/communicator/icons/turbo-systray.gif                     (communicator/icons/turbo-systray.gif)
   skin/classic/communicator/icons/btn1.gif                              (communicator/icons/btn1.gif)
   skin/classic/communicator/icons/smileys/smile_n.gif                   (communicator/icons/smileys/smile_n.gif)
   skin/classic/communicator/icons/smileys/frown_n.gif                   (communicator/icons/smileys/frown_n.gif)
--- mozilla/themes/classic/jar.mn.orig	2007-08-29 02:43:30.000000000 +0200
+++ mozilla/themes/classic/jar.mn	2007-08-29 02:44:12.000000000 +0200
@@ -49,10 +49,6 @@
   skin/classic/communicator/sidebar/sidebar.css                         (communicator/sidebar/sidebar.css)
   skin/classic/communicator/sidebar/sidebarListView.css                 (communicator/sidebar/sidebarListView.css)
   skin/classic/communicator/xpinstall/xpinstall.css                     (communicator/xpinstall/xpinstall.css)
-  skin/classic/communicator/brand/throbber-single.gif                   (communicator/brand/throbber-single.gif)
-  skin/classic/communicator/brand/throbber-anim.gif                     (communicator/brand/throbber-anim.gif)
-  skin/classic/communicator/brand/throbber16-single.gif                 (communicator/brand/throbber16-single.gif)
-  skin/classic/communicator/brand/throbber16-anim.gif                   (communicator/brand/throbber16-anim.gif)
   skin/classic/communicator/icons/offline.gif                           (communicator/icons/offline.gif)
   skin/classic/communicator/icons/online.gif                            (communicator/icons/online.gif)
   skin/classic/communicator/icons/search-act.gif                        (communicator/icons/search-act.gif)
@@ -62,7 +58,6 @@
   skin/classic/communicator/icons/lock-broken.gif                       (communicator/icons/lock-broken.gif)
   skin/classic/communicator/icons/lock-insecure.gif                     (communicator/icons/lock-insecure.gif)
   skin/classic/communicator/icons/loading.gif                           (communicator/icons/loading.gif)
-  skin/classic/communicator/icons/turbo-systray.gif                     (communicator/icons/turbo-systray.gif)
   skin/classic/communicator/icons/btn1.gif                              (communicator/icons/btn1.gif)
   skin/classic/communicator/icons/smileys/smile_n.gif                   (communicator/icons/smileys/smile_n.gif)
   skin/classic/communicator/icons/smileys/frown_n.gif                   (communicator/icons/smileys/frown_n.gif)
